src/handlers/input_resources/InputResourcesStreamHandler.h

```diff
@@ -0,0 +1,123 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#pragma once
+
+#include "AbstractMetricsManager.h"
+#include "DnstapInputStream.h"
+#include "MockInputStream.h"
+#include "PcapInputStream.h"
+#include "SflowInputStream.h"
+#include "StreamHandler.h"
+#include "ThreadMonitor.h"
+#include <Corrade/Utility/Debug.h>
+#include <limits>
+#include <string>
+
+namespace visor::handler::resources {
+
+using namespace visor::input::pcap;
+using namespace visor::input::dnstap;
+using namespace visor::input::mock;
+using namespace visor::input::sflow;
+
+constexpr double MEASURE_INTERVAL = 10; // in seconds
+
+class InputResourcesMetricsBucket final : public visor::AbstractMetricsBucket
+{
+
+protected:
+    mutable std::shared_mutex _mutex;
+
+    // total numPackets is tracked in base class num_events
+    Quantile<double> _cpu_percentage;
+    Quantile<uint64_t> _memory_usage_kb;
+    Counter _policies_number;
+    Counter _handlers_count;
+    bool _merged;
+
+public:
+    InputResourcesMetricsBucket()
+        : _cpu_percentage("resources", {"cpu_percentage"}, "Quantiles of thread cpu usage")
```

Comment:
let's make this "cpu_usage", both the metric name and our variables for consistency. description "Quantiles of 5s averages of CPU usage by the input stream"